handlers/graph.go

```diff
@@ -82,7 +83,7 @@ func graphNamespace(w http.ResponseWriter, r *http.Request, client *prometheus.C
 
 	log.Debugf("Build roots (root destination services nodes) for [%v] namespace graph with options [%+v]", o.Vendor, o)
 
-	trees := buildNamespaceTrees(o, client)
+	trees := buildNamespaceTrees(o, client, istioClient)
```

Comment:
Perhaps, instead to pass here the istioClient, it can be done as the following nodes.
Leave the trees unchanged, and add them on the following one.

Nothing change in the goal, but then, the logic for adding CB or any logic can be splitted and we can maintain the graph tests and add additional new tests on different tests files.

At least, that was the rationale for the graph_unused.go / graph_unused_test.go.

Of course this can be re-structured in N ways, but just thinking the easy one.

Reply:
I think this sounds like what @jshaughn  wanted to do, too. Build the graph and then later add to them additional "overlay" information (like this circuit breaker stuff).